### DISCUSSION TITLE
preview 1 to rc1 wierdness

### DISCUSSION THREAD

**Discussion by chrpai** [wixtoolset/issues#7095] [Questions]
I updated  heatwave and started playing with switching from preview 1 and I'm seeing some interesting behaviors.  I need to do more testing but maybe some of these will make sense just thinking about them.

Before updating heatwave or updating the msbuild reference I built a project expected to see the banner images default back to WiX Red but they didn't.  I got the correct images which wasn't supposed to happen until rc1.

On a different machine I installed Heatwave.  It seems the wixtoolkit sdk is "sticky".   For example

Package Project (preview 1)  references Module Project ( rc 1 )   the Package Project has the correct branding.
Package Project (preview 1)  references Module Project ( preview 1 ) the Package Project has the incorrect branding.

In both cases it shouldn't have the correct branding if Package Project was building using preview 1.  Now the MSBuild log starts to make sense:

>C:\GitHub\MusicKiosk\ProductName1\ProductName1\ProductName1.wixproj : warning MSB4240: Multiple versions of the same SDK "WixToolset.Sdk" cannot be specified. The previously resolved SDK version "4.0.0-rc.1" from location "C:\GitHub\MusicKiosk\ProductName1\ProductName1MM\ProductName1MM.wixproj" will be used and the version "4.0.0-preview.1" will be ignored.
2>C:\GitHub\MusicKiosk\ProductName1\ProductName1\ProductName1.wixproj : warning MSB4240: Multiple versions of the same SDK "WixToolset.Sdk" cannot be specified. The previously resolved SDK version "4.0.0-rc.1" from location "C:\GitHub\MusicKiosk\ProductName1\ProductName1MM\ProductName1MM.wixproj" will be used and the version "4.0.0-preview.1" will be ignored.

It looks like when a solution is built MSBuild can only be told to use one.  This could be a problem down the road when trying to go from WiX4 to WiX5.   We might need to include the version of WiX in the name like WiXToolset.SDK4  and WiXToolset.SDK5.

**Comment by robmen** [comment]
Did you delete the `obj` and `bin` folders? They cache information that can give this sort of error.

**Comment by chrpai** [comment]
Yes. Nothing conclusive yet... this is mostly a BOLO for multiple project scenarios.  I'm thinking most testing was probably done with a single project.
